### DISCUSSION TITLE
How to show all spectral locus?

### DISCUSSION THREAD

**Discussion by hminle** [colour-science/colour#1112] [Q&A] [answered]
Hi Colour team,

In this [function](https://colour.readthedocs.io/en/develop/generated/colour.plotting.plot_chromaticity_diagram_CIE1931.html), we show the chromaticity diagram with spectral locus, but some of them are missing (from 480 to 520).
How we show them on the diagram?
<img width="689" alt="image" src="https://user-images.githubusercontent.com/16201681/221453459-f932e759-2928-47df-871e-34223c19ec87.png">

Thanks

**Comment by thomasmansencal** [comment]
Hi @hminle,

You can pass custom labels by using `spectral_locus_labels`:

        Array of wavelength labels used to customise which labels will be drawn
        around the spectral locus. Passing an empty array will result in no
        wavelength labels being drawn.

https://github.com/colour-science/colour/blob/develop/colour/plotting/diagrams.py#L106

Cheers,

Thomas

**Reply by hminle** [reply]
Oh, I mean the default labels are correct, but it cannot be shown on the chart. Somehow the labels hide behind the axis.

**Reply by hminle** [reply]
Even if I turn off the axis, some labels cannot be shown
<img width="751" alt="image" src="https://user-images.githubusercontent.com/16201681/221454474-06791467-c313-479c-8994-1383acd533a7.png">

**Reply by thomasmansencal** [reply]
Oh you mean that they are cropped? If so you need to set the bounding box accordingly, try passing `bounding_box=[-0.1, 0.9, -0.1, 0.9]`.

Cheers,

Thomas

**Reply by hminle** [reply]
> 

Great, that solves my issue. Thank you so much Thomas

**Comment by hminle** [comment, marked as answer]
Here is the answer from Thomas: https://github.com/colour-science/colour/discussions/1112#discussioncomment-5117488
